server/src/main/java/io/crate/execution/dsl/projection/SourceIndexWriterReturnSummaryProjection.java

```diff
@@ -71,6 +76,7 @@ public SourceIndexWriterReturnSummaryProjection(RelationName relationName,
         sourceUri = (InputColumn) Symbols.fromStream(in);
         sourceUriFailure = (InputColumn) Symbols.fromStream(in);
         lineNumber = (InputColumn) Symbols.fromStream(in);
+        sourceParsingFailure = in.getVersion().before(Version.V_4_7_1) ? new InputColumn(3, DataTypes.STRING) : (InputColumn) Symbols.fromStream(in);
```

Comment:
I think the before 4.7.1 case deserves a comment. Why InputColumn 3?

Reply:
> why 3

was not clear to me whether I should follow the way I create it [here](https://github.com/crate/crate/pull/12229/files#diff-53c11b5d99dc6cd6d5ea9e7a33a2f3f197c476ddd9675737246257aedbce2468R259) - probably it's better to simply return null (with null check in another place consuming this Input)? https://github.com/crate/crate/pull/12229#discussion_r839255280

Reply:
I added a comment but haven't touched InputColumn - is streaming null a better idea? discussed [here](https://github.com/crate/crate/pull/12229#discussion_r839255280) with @jeeminso ?

Reply:
Good point, definitely need to add comments why 3. Also [implemented](https://github.com/crate/crate/pull/12229/commits/ac82a333e4a8e6682c1af600b76b006b6797582a) nullable approach - probably this one is slightly better as null is kinda self-documenting, at least no need to explain in 2 places pages of the specific index.